test/cancel_timer_test.ml

```diff
@@ -0,0 +1,34 @@
+[@@@warning "-8"]
+
+open Riot
+
+type Message.t += A | B
+
+let main () =
+  let (Ok _) = Logger.start () in
+  let this = self () in
+
+  (* We cancel the second timer before it sends any messages, so the only
+     message in the inbox should be A (sent from the first timer) *)
+  let (Ok _timer) = Timer.send_after this A ~after:20L in
+  let (Ok timer) = Timer.send_interval this B ~every:50L in
+  Timer.cancel timer;
```

Comment:
**question**. I think this would _always_ get you A before B since we are not canceling the first timer. So if we didn't cancel any timer, I'd expect the mailbox to fill like `[ A, B, B, B, ... ]` but now that we cancelled it, it should look like `[ A ]`.

We could fix this by making the `send_after` call wait _more_ than the first interval is sent.

Maybe it would be even easier to just do:

```ocaml
let (Ok _) = Timer.send_after this A ~after:100L in
let (Ok t) = Timer.send_after this B ~after:10L in
Timer.cancel t;
```

Reply:
Good catch!  I had misinterpreted `receive` and thought that it would return the newest message in the mailbox rather than the oldest.  I just pushed a commit that should fix this issue.

Thank you for the feedback, I appreciate it!